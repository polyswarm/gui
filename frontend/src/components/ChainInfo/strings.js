const strings = {
    current: 'Current Balance',
    adjusted: 'Balance after Transfer'
};
export default strings;