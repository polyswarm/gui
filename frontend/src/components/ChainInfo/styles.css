.ChainInfo {
    background-color: rgb(46, 255, 109);
    border-radius: 4px;
    box-shadow: 0 1px 2px #000000;
    box-sizing: border-box;
    color: #8500ff;
    display: flex;
    flex-direction: column;
    margin: 15px 0 15px 0;
    padding: 5px 20px 5px 20px;
    width: 45%;
}

.ChainInfo-Title > h2 {
    margin: 0 0 8px 0;
}

.ChainInfo-Content > ul > li {
    margin: 0 0 8px 0;
}

.ChainInfo-Content > ul {
    list-style: none;
    padding-left: 0;
}