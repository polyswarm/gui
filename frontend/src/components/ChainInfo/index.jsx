// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Project imports
// Component imports
import strings from './strings.js';
import './styles.css';

class ChainInfo extends Component {
  render() {
    const {props: {title, homeName, homeBalance, sideName, sideBalance}} = this;
    return (
        <div className='ChainInfo'>
          <div className='ChainInfo-Title'>
            <h2>{title}</h2>
          </div>
          <div className='ChainInfo-Content'>
            <p>{strings.balances}</p>
            <ul>
              <li><b>{homeName}</b>{': '}{homeBalance}</li>
              <li><b>{sideName}</b>{': '}{sideBalance}</li>
            </ul>
          </div>
        </div>
    )
  }
}

ChainInfo.proptypes = {
  title: PropTypes.string,
  homeName: PropTypes.string,
  homeBalance: PropTypes.string,
  sideName: PropTypes.string,
  sideBalance: PropTypes.string
}
export default ChainInfo;