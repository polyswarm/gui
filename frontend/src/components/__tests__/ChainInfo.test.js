import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import ChainInfo from '../ChainInfo';

it('renders without crashing', () => {
const wrapper = render(<ChainInfo />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});