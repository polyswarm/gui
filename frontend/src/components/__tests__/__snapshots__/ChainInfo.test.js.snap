// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="ChainInfo"
>
  <div
    class="ChainInfo-Title"
  >
    <h2>
      
    </h2>
  </div>
  <div
    class="ChainInfo-Content"
  >
    <p>
      
    </p>
    <ul>
      <li>
        <b>
          
        </b>
        : 
      </li>
      <li>
        <b>
          
        </b>
        : 
      </li>
    </ul>
  </div>
</div>
`;
